core/src/main/java/io/confluent/kafka/schemaregistry/storage/DeleteSubjectKey.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2014 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.kafka.schemaregistry.storage;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import org.codehaus.jackson.annotate.JsonPropertyOrder;
+import org.hibernate.validator.constraints.NotEmpty;
+
+@JsonPropertyOrder(value = {"keytype", "subject", "magic"})
+public class DeleteSubjectKey extends SchemaRegistryKey {
+
+  private static final int MAGIC_BYTE = 0;
+  @NotEmpty
+  private String subject;
+
+  public DeleteSubjectKey(@JsonProperty("subject") String subject) {
```

Comment:
Hmm, I guess with the `JsonPropertyOrder` annotation it'll extract private fields as well.

I worry about this because I don't know that we actually verify the format of the data written to Kafka. And we definitely don't have upgrade tests -- there's only been one version of the format.

In fact, since we added a field to `SchemaValue`, it's possible we should be bumping the magic number for `SchemaKey`/`SchemaValue`. (I'm actually not sure why we only put it in one of them, though the idea may have been similar to how Kafka handles request/response versioning.) In this case it is backwards compatible and not modifying it would allow us to avoid adding extra configs to control which version is to be used, more logic around handling multiple versions, etc, though it's a bit risky if we don't test compatibility around upgrades...